hicommonwealth/commonwealth
# Enable community stake in create community flow

## Link to Issue
Closes: #5878

## Description of Changes
- moved chainConfig to `libs/chains/src/commonProtocol` to be able to use it both in FE and BE
- removed duplicated files in `ContractHelpers` (removed `.tsx`, remained `.ts`)
- few commits from @ianrowan for fixing protocol layer
- created flow and hooked it up to the API for third step in create community flow


## Test Plan
### Prerequisites 
- add `FLAG_COMMUNITY_STAKE=true` to `.env` in order to enable third step in create community flow
- add `ETH_RPC=test` to `.env` to prevent metamask from asking to switch network back from `goerli` to `mainnet`
- prepare metamask wallet that is switched to `goerli` network and has positive balance (for top up use faucets: [one](https://goerlifaucet.com/), [two](https://faucet.quicknode.com/ethereum/goerli) or [three](https://goerli-faucet.pk910.de/))

### Testing
- go to create community flow
- there should be only 2 steps visible on top
- choose evm
- -----
- third step should be visible only if you choose `goerli` from chain dropdown
- click launch community
- -----
-  in `Do you want to enable community stake?` namespace and symbol field should be auto-populated based on previous step
- check validation for both fields 
- clicking `No` should move user to success screen
-  clicking `Yes` should move user to next page within 3rd step (and run additional validation for namespace field)
- -----
- in `Sign transactions to launch stakes` only first `sign` button should be visible on start
- on clicking `cancel` popup should appear (opting out should take user to success screen)
- on clicking first `Sign`, metamask extension should ask for signing transaction and then the spinner should be visible and cancel button should be disabled
- after success transaction, first button should be disabled and second should be active
- on clicking second `sign`, metamask extension should ask for signing transaction and then the spinner should be visible and cancel button should be disabled
- after success transaction, user should be redirected to success screen right away
- -----
- as admin, you can go to community profile page and field namespace should be populated with namespace you typed in the create community flow
- -----
video of the flow

https://github.com/hicommonwealth/commonwealth/assets/14819225/49660a8e-a3fa-4092-9fb6-301ff1bfd715


## Deployment Plan

1. add `FLAG_COMMUNITY_STAKE=true`
2. ⚠️ **for now it works only in goerli testnet, we need to deploy contracts to eth mainnet before production. More info should be provided from @ianrowan**
3. there are some changes in the code necessary to switch goerli to mainnet, not only in this PR but also in several places in the app
